### ISSUE TITLE
After last updates, ForneyLab cannot build message passing algorithm.

### ISSUE THREAD

**Issue by albertpod** [biaslab/ForneyLab.jl#132]
After the commit `61aba6fef0b06b12cbba177c60089a17d8ebd295 (Renamed update rules for Equality node.)`, ForneyLab fails to build the algorithm for the following state-space model:

```
z_t ~ N(z_{t-1}, \gamma^{-1})
x_t ~ N(z_{t}, \tau^{-1})
y_t ~ N(z_{t}, \omega*\tau^{-1})
```
The corresponding code:
```julia
using ForneyLab

# Building the model
n_samples = 1
fg = FactorGraph()

# State prior
@RV z_0 ~ GaussianMeanVariance(placeholder(:m_z_0), placeholder(:v_z_0))

@RV γ ~ Gamma(placeholder(:a_γ), placeholder(:b_γ))

@RV τ ~ Gamma(placeholder(:a_τ), placeholder(:b_τ))

# Transition
z = Vector{Variable}(undef, n_samples)
# Input
x = Vector{Variable}(undef, n_samples)
# Output
y = Vector{Variable}(undef, n_samples)
# Intervention
ω = Vector{Variable}(undef, n_samples)

z_i_min = z_0
for i in 1:n_samples

    @RV z[i] ~ GaussianMeanPrecision(z_i_min, γ)

    @RV x[i] ~ GaussianMeanPrecision(z[i], τ)

    @RV ω[i]

    @RV y[i] ~ GaussianMeanPrecision(z[i], ω[i]*τ)


    # Data placeholder
    placeholder(y[i], :y, index=i)
    placeholder(ω[i], :ω, index=i)
    placeholder(x[i], :x, index=i)

    # Reset state for next step
    z_i_min = z[i]
end

q = PosteriorFactorization(z_0, z, γ, τ, ids=[:Z0 :Z :Γ :T])
algo = messagePassingAlgorithm(free_energy=true)
```

The error at the aforementioned  commit:
```julia
LoadError: No applicable SumProductRule{Multiplication} update for Multiplication node with inbound types: Message{Union{Gamma, Wishart},var_type} where var_type<:ForneyLab.VariateType, Nothing, Message{PointMass,var_type} where var_type<:ForneyLab.VariateType
```

**Comment by ivan-bocharov** [comment]
Actual error related to the issue is:

```julia
LoadError: Multiple applicable SumProductRule{Equality} updates for Equality node with inbound types: Message{Union{Gamma, Wishart},var_type} where var_type<:ForneyLab.VariateType, Nothing, Message{Gamma,var_type} where var_type<:ForneyLab.VariateType
```

This happens because of the newly introduced rules for MP around Equality node. I'm working on a fix.

**Comment by ivan-bocharov** [comment]
Resolved by a03176d5a6677c9eb67ec23f27c732ac6defea5e.
